src/web/resources/report.js

```diff
@@ -35,6 +35,94 @@ function Element(tagname, props, children) {
     return $elt;
 }
 
+var renames = {};
+renames["imp-start"] = "Improved start"
+renames["apx-start"] = "Approximate start"
+renames["uni-start"] = "Regressed from start"
+renames["ex-start"]  = "Exact start"
+renames["eq-start"]  = "Equal start"
+renames["lt-start"]  = "Less than start"
+renames["gt-start"]  = "Greater than start"
+renames["gt-target"] = "Greater than target"
+renames["eq-target"] = "Equal than target"
+renames["lt-target"] = "Less than target"
+renames["error"]     = "Error"
+renames["timeout"]   = "Timeout"
+renames["crash"]     = "Crash"
+
+const Filters = new Component("#filters", {
+    setup: function () {
+        // Warning fragile starting state
+        const regressedTags = ["uni-start", "lt-target", "lt-start", 
+        "apx-start", "timeout", "crash", "error"]
+        const improvedTags = ["imp-start", "ex-start", "eq-start", "eq-target",
+        "gt-target"]
+        const advancedTags = []
+        advancedTags.push(regressedTags)
+        advancedTags.push(improvedTags)
+
+        // build labels
+        const regressedLabel = this.buildCheckboxLabel("regressed", "Regressed", true)
+        const improvedLabel = this.buildCheckboxLabel("improved", "Improved", true)
+
+        // add listeners
+        regressedLabel.addEventListener("click", this.attachLeaderToChildren("regressed", regressedTags))
+        improvedLabel.addEventListener("click", this.attachLeaderToChildren("improved", improvedTags))
+
+        const improvedChildren = improvedTags.map((child) => {
+            return this.createChild(child)
+        })
+        const regressedChildren = regressedTags.map((child) => {
+            return this.createChild(child)
+        })
+        const advancedDiv = [improvedChildren, regressedChildren]
+
+        this.elt.appendChild(Element("div", ["Filters"]))
+        this.elt.appendChild(Element("div", [improvedLabel, regressedLabel]))
+        this.elt.appendChild(Element("details", [Element("summary", "Advanced"), advancedDiv]))
+    },
+    buildCheckboxLabel: function(idTag, text, boolState) {
+        return Element("label", { id: idTag }, [
+            Element("input", { type: "checkbox", checked: boolState }, ""),
+            text])
+    },
+    attachLeaderToChildren: function (leaderTag,listOfTags) {
+        return () => {
+            const improvedBox = document.querySelector(`#${leaderTag} input`)
+            listOfTags.forEach((str) => {
+                const currentCheckBox = document.querySelector(`#${str} input`)
+                currentCheckBox.checked = improvedBox.checked
+                this.updateDomNodesWithID(`${str}`, currentCheckBox.checked)
+            })
+        }
+    },
+    createChild: function (childName) {
+        const count = document.querySelectorAll(`tr.${childName}`)
+        const childBox = this.buildCheckboxLabel(childName, `${renames[childName]} (${count.length})`, true)
```

Comment:
Probably better if the count was a separate argument, in case we later want to wrap it in some tag or whatever.

Reply:
And then you wouldn't need so much string interpolation either.